src/notation/view/pianokeyboard/pianokeyboardpanelcontextmenumodel.cpp

```diff
@@ -167,6 +203,35 @@ MenuItem* PianoKeyboardPanelContextMenuModel::makeNumberOfKeysItem(const muse::T
     return item;
 }
 
+MenuItem* PianoKeyboardPanelContextMenuModel::makeToggleNotatedPitchItem(const muse::TranslatableString& title, bool state)
+{
+    UiAction action;
+    action.title = title;
+    action.code = SET_NOTATED_PITCH_CODE;
+    action.checkable = Checkable::Yes;
+
+    MenuItem* item = new MenuItem(action, this);
+    if (state) {
+        item->setId(QString::number(129)); // Just to not conflict with other buttons
+    } else {
+        item->setId(QString::number(130));
+    }
```

Comment:
I'd go for a more meaningful ID, since that is more future-proof:
```suggestion
    item->setId(QString::fromStdString(SET_NOTATED_PITCH_CODE) + (state ? "-written" : "-concert"));
```